### PR TITLE
Remove CoffeeScript

### PR DESCRIPTION
**Pull request by aeaton-overleaf**
* Decaffeinate the main module using `bulk-decaffeinate` and some manual tidying.
* Remove the `coffeescript` dependency.
* Bump the version number to `2.0.0`.
* Rename the package to `@overleaf/settings-module`.

This will now throw an error if a `.coffee` settings file is encountered, instead of the previous deprecation warning. At some point this can be removed, but I think it's useful to make sure settings aren't accidentally ignored.

### CODE REVIEW

**Review by das7pad — CHANGES_REQUESTED** [review]
I'm not sure about the removal of coffee-script support for custom settings files as this will affect all Server Pro customers/Server CE users.

Have you thought about lazy loading coffee-script when detecting a `.coffee` file?

**Comment by aeaton-overleaf** [comment]
> I'm not sure about the removal of coffee-script support for custom settings files as this will affect all Server Pro customers/Server CE users

The point of this is to remove the need for CoffeeScript, so I'm not sure…

**Comment by jdleesmiller** [comment]
> will affect all Server Pro customers/Server CE users.

We should definitely mention this in the release notes, and we can link to tools like decaffeinate to help, but I think we should push forward with removing it. (And this shouldn't affect most people, because a lot of the config is via env vars, which will continue to work.)

File: package.json [hunk 1]
```diff
@@ -1,13 +1,6 @@
...
-  "description": "A centralised settings system for ShareLaTeX",
-  "version": "1.3.0",
-  "repository": {
-    "type": "git",
-    "url": "https://github.com/sharelatex/settings-sharelatex.git"
-  },
-  "dependencies": {
-    "coffee-script": "1.6.0"
-  }
+  "name": "@overleaf/settings-module",
```
Comment (das7pad) [inline package.json:2]: We are omitting the `-module` suffix in our other modules in the `@overleaf` namespace. WDYT about omitting it here as well? https://www.npmjs.com/search?q=%40overleaf

Comment (aeaton-overleaf) [reply package.json:2]: I thought about that, but `@overleaf/settings` maybe implies that it's the actual settings?

I don't mind much either way, but this repo should probably also be renamed if the name is going to change.

Comment (das7pad) [reply package.json:2]: > I thought about that, but `@overleaf/settings` maybe implies that it's the actual settings?

Upon importing it you do get the actual config of the current service ;)

Comment (aeaton-overleaf) [reply package.json:2]: That's true 😄

Comment (das7pad) [reply package.json:2]: Yeah, naming is hard. We've got
- `redis-wrapper` -> `@overleaf/redis-wrapper`
-  `metrics-module` -> `@overleaf/metrics`
-  `o-error` -> `@overleaf/o-error`

WDYT about a poll in slack what others think?

Comment (aeaton-overleaf) [reply package.json:2]: Either a poll or I'm happy for @jdleesmiller to weigh in. The precedent of `metrics-module` -> `@overleaf/metrics` implies that `settings-module` -> `@overleaf/settings` would be fine, so maybe let's go for that?

Comment (jdleesmiller) [reply package.json:2]: > let's go for that?

OK by me! I am persuaded by https://github.com/overleaf/settings-module/pull/5#discussion_r640513093 :)

**Comment by aeaton-overleaf** [comment]
The name is now `@overleaf/settings`

**Review by das7pad — APPROVED** [review]

**Pull request merged**